src/main/java/com/minecolonies/coremod/network/messages/ReplaceBlockMessage.java

```diff
@@ -0,0 +1,202 @@
+     * @param blockTo the block to replace it with.
+     */
+    public ReplaceBlockMessage(@NotNull final BlockPos pos1, @NotNull final BlockPos pos2, @NotNull final ItemStack blockFrom, @NotNull final ItemStack blockTo)
+    {
+        super();
+        this.from = pos1;
+        this.to = pos2;
+        this.blockFrom = blockFrom;
+        this.blockTo = blockTo;
+    }
+
+    @Override
+    public void fromBytes(@NotNull final ByteBuf buf)
+    {
+        from = BlockPosUtil.readFromByteBuf(buf);
+        to = BlockPosUtil.readFromByteBuf(buf);
+        blockTo = ByteBufUtils.readItemStack(buf);
+        blockFrom = ByteBufUtils.readItemStack(buf);
+    }
+
+    @Override
+    public void toBytes(@NotNull final ByteBuf buf)
+    {
+        BlockPosUtil.writeToByteBuf(buf, from);
+        BlockPosUtil.writeToByteBuf(buf, to);
+        ByteBufUtils.writeItemStack(buf, blockTo);
+        ByteBufUtils.writeItemStack(buf, blockFrom);
+    }
+
+    @Override
+    public void messageOnServerThread(final ReplaceBlockMessage message, final EntityPlayerMP player)
+    {
+        final World world = player.getServerWorld();
+        final FakePlayer fakePlayer = new FakePlayer(player.getServerWorld(), new GameProfile(player.getUniqueID(), "placeStuffForMePl0x"));
+        for(int x = Math.min(message.from.getX(), message.to.getX()); x <= Math.max(message.from.getX(), message.to.getX()); x++)
+        {
+            for (int y = Math.min(message.from.getY(), message.to.getY()); y <= Math.max(message.from.getY(), message.to.getY()); y++)
+            {
+                for (int z = Math.min(message.from.getZ(), message.to.getZ()); z <= Math.max(message.from.getZ(), message.to.getZ()); z++)
+                {
+                    final BlockPos here = new BlockPos(x, y, z);
+                    final IBlockState blockState = world.getBlockState(here);
+                    final ItemStack stack = BlockUtils.getItemStackFromBlockState(blockState);
+                    if (correctBlockToRemoveOrReplace(stack, blockState, message.blockFrom))
+                    {
+                        if ((blockState.getBlock() instanceof BlockDoor && blockState.getValue(BlockDoor.HALF) == BlockDoor.EnumDoorHalf.UPPER)
+                                || (blockState.getBlock() instanceof BlockBed && blockState.getValue(BlockBed.PART) == BlockBed.EnumPartType.HEAD))
+                        {
+                            continue;
+                        }
+
+                        world.setBlockToAir(here);
+                        final ItemStack stackToPlace = message.blockTo.copy();
+                        stackToPlace.setCount(stackToPlace.getMaxStackSize());
+                        fakePlayer.setHeldItem(EnumHand.MAIN_HAND, stackToPlace);
+
+                        if (message.blockTo.getItem() instanceof ItemBed)
+                        {
+                            fakePlayer.rotationYaw = blockState.getValue(BlockBed.FACING).getHorizontalIndex() * 90;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 90 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)